instrumentation/async-http-client-1.9/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/asynchttpclient/ResponseAdvice.java

```diff
@@ -26,14 +25,14 @@ public static Scope onEnter(
     // After response was handled by user provided handler.
     ContextStore<AsyncHandler, Pair> contextStore =
         InstrumentationContext.get(AsyncHandler.class, Pair.class);
-    Pair<Context, Span> spanWithParent = contextStore.get(handler);
-    if (null != spanWithParent) {
+    Pair<Context, Context> contextWithParent = contextStore.get(handler);
+    if (contextWithParent != null) {
       contextStore.put(handler, null);
     }
-    if (spanWithParent.hasRight()) {
-      AsyncHttpClientTracer.tracer().end(spanWithParent.getRight(), response);
+    if (contextWithParent.hasRight()) {
```

Comment:
Can `contextWithParent` be null or not? These two `ifs` clearly disagrees on this.